pkg/capi/gcp.go

```diff
@@ -0,0 +1,294 @@
+package capi
+
+import (
+	"context"
+	"fmt"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	configv1 "github.com/openshift/api/config/v1"
+	mapiv1 "github.com/openshift/api/machine/v1beta1"
+	framework "github.com/openshift/cluster-api-actuator-pkg/pkg/framework"
+	corev1 "k8s.io/api/core/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	gcpv1 "sigs.k8s.io/cluster-api-provider-gcp/api/v1beta1"
+	clusterv1 "sigs.k8s.io/cluster-api/api/v1beta1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/envtest/komega"
+	yaml "sigs.k8s.io/yaml"
+)
+
+const (
+	infraAPIVersion        = "infrastructure.cluster.x-k8s.io/v1beta1"
+	gcpMachineTemplateName = "gcp-machine-template"
+)
+
+var (
+	ctx         = context.Background()
+	clusterName string
+	cl          client.Client
+)
+
+var _ = Describe("Cluster API GCP MachineSet", Ordered, func() {
```

Comment:
Please add `framework.LabelCAPI, framework.LabelDisruptive` label so that we can select cases in qe jobs and our pre-submit jobs for rebase.